### ISSUE TITLE
doom-modeline now uses nerdfonts

### ISSUE THREAD

**Issue by andresilva** [doomemacs/doomemacs#7368]
### I confirm that...

- [X] I have searched the [issue tracker](https://github.com/doomemacs/doomemacs/issues), [documentation](https://docs.doomemacs.org), [FAQ](https://docs.doomemacs.org/-/faq), [Discourse](https://discourse.doomemacs.org), and [Google](https://google.com), in case this issue has already been reported/resolved.

- [X] I have read "[How to Debug Issues](https://doomemacs.org/d/how2debug)", and will use it to provide as much information about this issue as possible.

- [X] The issue can be reproduced on the **latest** available commit of Doom.

- [X] The issue can be reproduced on a stable release of Emacs, such as 27 or 28. *(Doom does not support development builds like 29+ or any version ending in .50 or .9x)*

### Expected behavior

After installing doomemacs I should have working icons on the modeline.

### Current behavior

Icons are currently not working since doom-modeline has migrated to use nerdfonts instead of all-the-icons font with the latest release (https://github.com/seagle0128/doom-modeline/pull/622). The installer should probably prompt for installing this font now instead of all-the-icons (although I'm not sure if that's still used by doomemacs itself).

### Steps to reproduce

Self-descriptive

### System Information

---

**Comment by philiparvidsson** [comment]
Meanwhile, how can I fix this manually?

**Comment by hvzzzz** [comment]
I have disabled all-the-icons package and the icons in doom-modeline have returned to normal.
Just add
`(package! all-the-icons :disable t)`
To your packages.el file

**Comment by nnungest** [comment]
I think this is the commit that introduced those changes https://github.com/doomemacs/doomemacs/commit/7a5f5d07e72d496ff614e0892c78b62b31aaac4d
 
For me I had mixed results and just decided to `git checkout` to a commit prior to the UI bump changes. Hope this helps others.

**Comment by finex** [comment]
Thanks for the information. It should be written somewhere.

**Comment by ebeem** [comment]
I think doom still uses `all-the-icons` in many other packages including `doom-ui` and `doom-dashboard`
disabling `all-the-icons` caused dependency bugs in the above packages to me.
I am not exactly sure why I have to disable it for `doom-modeline` to work again even though I already have `nerdfonts` installed

To get it working properly, I pinned `doom-modeline` just before `nerdfonts` was introduced.

`packages.el`
````elisp
(package! doom-modeline :pin "918730eff72e")
`````

**Comment by lorenzohess** [comment]
I also had this problem after upgrading Doom.

I noticed I had set a custom font in my `config.el` and commented out that line:
```elisp
;; (set-face-attribute 'mode-line nil :font "Mononoki Nerd Font")
```

Now the modeline icons and all other icons (e.g. in `dired`) **look normal**. I do have both `all-the-icons` and `nerd-icons` installed.

**Comment by hrqmonteiro** [comment]
> 

Not working:

![image](https://github.com/doomemacs/doomemacs/assets/17055027/312aade2-bd5b-4cf5-a7fc-4d03cb2352a0)

**Comment by hrqmonteiro** [comment]
> I think doom still uses `all-the-icons` in many other packages including `doom-ui` and `doom-dashboard` disabling `all-the-icons` caused dependency bugs in the above packages to me. I am not exactly sure why I have to disable it for `doom-modeline` to work again even though I already have `nerdfonts` installed
> 
> To get it working properly, I pinned `doom-modeline` just before `nerdfonts` was introduced.
> 
> `packages.el`
> 
> ```emacs-lisp
> (package! doom-modeline :pin "918730eff72e")
> ```

Also not working.

**Comment by hrqmonteiro** [comment]
> > > I think doom still uses `all-the-icons` in many other packages including `doom-ui` and `doom-dashboard` disabling `all-the-icons` caused dependency bugs in the above packages to me. I am not exactly sure why I have to disable it for `doom-modeline` to work again even though I already have `nerdfonts` installed
> > > To get it working properly, I pinned `doom-modeline` just before `nerdfonts` was introduced.
> > > `packages.el`
> > > ```emacs-lisp
> > > (package! doom-modeline :pin "918730eff72e")
> > > ```
> > 
> > 
> > Also not working.
> 
> Have you followed the steps probably? I think it's needless to say you need to perform `doom upgrade` and restart emacs after the above changes

Yes i did that. Still showing those errors.

I also figured it could probably be cause i was setting treemacs with all-the-icons, and i even disabled that. Still showing the errors.

**Comment by hrqmonteiro** [comment]
> > > I think doom still uses `all-the-icons` in many other packages including `doom-ui` and `doom-dashboard` disabling `all-the-icons` caused dependency bugs in the above packages to me. I am not exactly sure why I have to disable it for `doom-modeline` to work again even though I already have `nerdfonts` installed
> > > To get it working properly, I pinned `doom-modeline` just before `nerdfonts` was introduced.
> > > `packages.el`
> > > ```emacs-lisp
> > > (package! doom-modeline :pin "918730eff72e")
> > > ```
> > 
> > 
> > Also not working.
> 
> Have you followed the steps probably? I think it's needless to say you need to perform `doom upgrade` and restart emacs after the above changes

Now i reinstalled Doom Emacs with the default config and did that and got this error:

![image](https://github.com/doomemacs/doomemacs/assets/17055027/16521115-aa8b-4bf1-8daf-5af40121029c)

**Comment by ebeem** [comment]
> > > > I think doom still uses `all-the-icons` in many other packages including `doom-ui` and `doom-dashboard` disabling `all-the-icons` caused dependency bugs in the above packages to me. I am not exactly sure why I have to disable it for `doom-modeline` to work again even though I already have `nerdfonts` installed
> > > > To get it working properly, I pinned `doom-modeline` just before `nerdfonts` was introduced.
> > > > `packages.el`
> > > > ```emacs-lisp
> > > > (package! doom-modeline :pin "918730eff72e")
> > > > ```
> > > 
> > > 
> > > Also not working.
> > 
> > Have you followed the steps probably? I think it's needless to say you need to perform `doom upgrade` and restart emacs after the above changes
> 
> Now i reinstalled Doom Emacs with the default config and did that and got this error:
> 
> ![image](https://github.com/doomemacs/doomemacs/assets/17055027/16521115-aa8b-4bf1-8daf-5af40121029c)
> 

I think you disabled all-the-icons and it's not installed anymore, if you implemented other workarounds that ignore all-the-icons then delete them

**Comment by hrqmonteiro** [comment]
> > > > > I think doom still uses `all-the-icons` in many other packages including `doom-ui` and `doom-dashboard` disabling `all-the-icons` caused dependency bugs in the above packages to me. I am not exactly sure why I have to disable it for `doom-modeline` to work again even though I already have `nerdfonts` installed
> > > > > To get it working properly, I pinned `doom-modeline` just before `nerdfonts` was introduced.
> > > > > `packages.el`
> > > > > ```emacs-lisp
> > > > > (package! doom-modeline :pin "918730eff72e")
> > > > > ```
> > > > 
> > > > 
> > > > Also not working.
> > > 
> > > 
> > > Have you followed the steps probably? I think it's needless to say you need to perform `doom upgrade` and restart emacs after the above changes
> > 
> > 
> > Now i reinstalled Doom Emacs with the default config and did that and got this error:
> > ![image](https://user-images.githubusercontent.com/17055027/264676242-16521115-aa8b-4bf1-8daf-5af40121029c.png)
> 
> I think you disabled all-the-icons and it's not installed anymore, if you implemented other workarounds that ignore all-the-icons then delete them

That's the vanilla doomemacs config, i literally haven't changed anything!

**Comment by hrqmonteiro** [comment]
I reinstalled Doom Emacs with the default config and only tried the solution you've mentioned:

![image](https://github.com/doomemacs/doomemacs/assets/17055027/dad817da-320d-4e31-969e-391124aaad26)

Still not working. And that is with the default installation. This is kinda strange. How can this be the default behavior?

**Comment by hrqmonteiro** [comment]
> @hrqmonteiro just made a fresh installation and downloaded all packages over. it worked fine.
> 
> Just in case, it will be a good idea to delete the folder `~/.emacs.d` or `~/.config/emacs` (depending on your configuration) before performing fresh installation.
> 
> Also note that emacs by default points to `.emacs.d` and if it wasn't found it points to `~/.config/emacs` In doom's readme now it clones to `~/.config/emacs` , so make sure you delete `~/.emacs.d` if it exists

I don't have a `.emacs.`.

My doom-modeline is still broken, not showing correct icons

![image](https://github.com/doomemacs/doomemacs/assets/17055027/4006520d-e648-4813-a722-2d4e6f8a34ec)

Is there no one to help? Really?

In other places, like company and centaur-tabs everytihng is fine:

![image](https://github.com/doomemacs/doomemacs/assets/17055027/f126ce82-223f-4b06-b8ad-da8c54856512)
![image](https://github.com/doomemacs/doomemacs/assets/17055027/ed15da8d-5186-454f-acfe-8c708ac7e92e)

**Comment by hlissner** [comment]
To summarize the issue:

Recent versions of `doom-modeline` have replaced [all-the-icons](<https://github.com/domtronn/all-the-icons.el>) with [nerd-icons](<https://github.com/rainstormstudio/nerd-icons.el>), which depends on a whole new set of fonts. Doom's upgrade/update process does not take this into account, and will not install those fonts for you at the moment. 

I'm considering dropping all-the-icons Doom-wide (for nerd-icons) as well, since the latter is better supported in more cases (like the terminal). But until that happens, you'll have to install these new fonts yourself or revert to a version of `doom-modeline` predating this change.

### TL;DR solution
Either:

1. Pin `doom-modeline` to a commit predating the change, i.e. Add `(package! doom-modeline :pin "918730eff72e155cfb31b012493a7f425bc48ff8")` to `$DOOMDIR/packages.el`, run `doom sync -u` in the terminal (the `-u` is important when changing `package!` recipes and pins), then restart Emacs.

2. Or run `M-x nerd-icons-install-fonts` to fix the problem. For Windows users only: the command will only download the fonts, you'll have to install them manually afterwards.

   If you opt for #2 and still have issues, try unpinning `doom-modeline` to ensure you have the latest version (by adding `(unpin! doom-modeline)` to `$DOOMDIR/packages.el` and running `doom sync -u`).

   If this *still* doesn't resolve the issue, try installing the nerd fonts manually [from the nerd-icons.el repo](https://github.com/rainstormstudio/nerd-icons.el/tree/main/fonts) or from https://www.nerdfonts.com/font-downloads (particularly "Symbols Nerd Font Mono").

**Comment by AlexDaniel** [comment]
OK, I have gotten into this (somewhat unrelated) problem twice now after trying to upgrade my doom setup, so I'll leave a message here if I get into it the third time because I just can't remember the fix every time.

I was getting this error:
```bash
doom sync -u
```
```
! Failed to locate a ':tools gist' module
- Using Emacs 29.3 @ /usr/bin/emacs
> Synchronizing "default" profile...
  > Regenerating envvars file
    ✓ Generated ~/.config/emacs/.local/env
  x There was an unexpected runtime error
    Message: Could not find package nerd-icons. Updating recipe repositories: (org-elpa melpa nongnu-elpa gnu-elpa-mirror el-get emacsmirror-mirror) with ‘straight-pull-recipe-repositories’ may fix this
    Backtrace:
      (error "Could not find package %S. Updating recipe repositories: %S with `straight-pull-recipe-repositories' may fix this" nerd-icons (org-elpa melpa nongnu-elpa gnu-elpa-mirror el-get emacsmirror-mirror))
      (if (straight--package-built-in-p melpa-style-recipe) (throw '--cl-block-straight--convert-recipe-- (list ':type 'built-in ':package (symbol-name melpa-style-recipe))) (error (concat "Could not find package %S. " "Updating recipe ...
      (or (straight-recipes-retrieve melpa-style-recipe nil cause) (if (straight--package-built-in-p melpa-style-recipe) (throw '--cl-block-straight--convert-recipe-- (list ':type 'built-in ':package (symbol-name melpa-style-recipe))) (...
      (if recipe-specified-p melpa-style-recipe (or (straight-recipes-retrieve melpa-style-recipe nil cause) (if (straight--package-built-in-p melpa-style-recipe) (throw '--cl-block-straight--convert-recipe-- (list ':type 'built-in ':pa...
      (let* ((recipe-specified-p (listp melpa-style-recipe)) (full-melpa-style-recipe (if recipe-specified-p melpa-style-recipe (or (straight-recipes-retrieve melpa-style-recipe nil cause) (if (straight--package-built-in-p melpa-style-r...
      (or (and (symbolp melpa-style-recipe) (gethash (symbol-name melpa-style-recipe) straight--recipe-cache)) (let* ((recipe-specified-p (listp melpa-style-recipe)) (full-melpa-style-recipe (if recipe-specified-p melpa-style-recipe (or...
      (catch '--cl-block-straight--convert-recipe-- (if (memq melpa-style-recipe straight-built-in-pseudo-packages) (progn (throw '--cl-block-straight--convert-recipe-- (list ':type 'built-in ':package (symbol-name melpa-style-recipe)))...
      (straight--convert-recipe nerd-icons nil)
      (let ((recipe (straight--convert-recipe (or (straight--get-overridden-recipe (if (listp melpa-style-recipe) (car melpa-style-recipe) melpa-style-recipe)) melpa-style-recipe) cause))) (straight--register-recipe recipe) (let* ((--cl...
      (catch '--cl-block-straight-use-package-- (let ((recipe (straight--convert-recipe (or (straight--get-overridden-recipe (if (listp melpa-style-recipe) (car melpa-style-recipe) melpa-style-recipe)) melpa-style-recipe) cause))) (stra...
      (straight-use-package nerd-icons no-clone)
      (straight-register-package nerd-icons)
    ! Wrote extended backtrace to ~/.config/emacs/.local/state/logs/cli.doom.240510202428.56488.error
  ! Script was abruptly aborted, leaving Doom in an incomplete state!
  - Run 'doom sync' to repair it.
```

And the fix is:
```bash
rm -rf ~/.config/emacs/.local/straight/
```
(or `~/.emacs.d/.local/straight/` if that's where your stuff is)

And then run `doom sync` and wait for a long time.

The answer was given here: https://discord.com/channels/406534637242810369/1155946544579760299/1158123908575264940
